lib/src/rules/implicit_reopen.dart

```diff
@@ -0,0 +1,229 @@
+// Copyright (c) 2023, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/dart/ast/visitor.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/type.dart';
+
+import '../analyzer.dart';
+
+const _desc = r"Don't implicitly reopen classes or mixins";
+
+/// todo(pq): link out to the spec or dart.dev docs.
```

Comment:
I wouldn't link out to the spec because I think most users find specifications challenging to read.